src/Implement/ResponseEncoder.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OperateData.h"
+#include "ResponseEncoder.h"
+
+ResponseEncoder* only_response_encoder = nullptr;
+
+ResponseEncoder* ResponseEncoder::GetInstance() {
+  if (only_response_encoder == nullptr) {
+    only_response_encoder = new ResponseEncoder();
+  }
+  return only_response_encoder;
+}
+
+std::string ResponseEncoder::Encode(const std::unique_ptr<OperateData>& data) {
+    
+    std::string temp_string = "HTTP/1.1 " + data->header["status_code"] + " " + data->header["status"] + '\n';
```

Comment:
`result_response` might be a better naming.
Usually for large amount of string concatenation, StringBuilder or similar is used in other language. For C++, it is optional because C++ string can be resized.